python/tank/deploy/tank_command.py

```diff
@@ -47,9 +47,9 @@
 ###############################################################################################
 # Built in actions (all in the tank_commands sub module)
 
-BUILT_IN_ACTIONS = [setup_project.SetupProjectAction, 
+BUILT_IN_ACTIONS = [setup_project.SetupProjectAction,
                     setup_project_wizard.SetupProjectFactoryAction,
-                    core_upgrade.CoreUpgradeAction, 
+                    core_upgrade.CoreUpdateAction,
```

Comment:
I might have missed something and/or things may have changed over time, but heads up that this was originally specifically designed to *not* be part of the public API (C++ pimpl style). When you use the tk API command factory, you get a `SgtkSystemCommand` back that you use to interact with the underlying tk command implementation:

```python
>>> tk
<Sgtk Core v0.16.42@0x102c7e810 Config /mnt/wintermute/configs/sha_is_in_the_heart>
>>> tk.list_commands()
['setup_project', 'setup_project_factory', 'core', 'localize', 'share_core', 'attach_to_core', 'validate', 'cache_apps', 'clear_shotgun_menu_cache', 'app_info', 'install_app', 'install_engine', 'updates', 'configurations', 'synchronize_folders', 'unregister_folders', 'clone_configuration', 'copy_apps', 'cache_yaml', 'get_entity_commands']
>>> tk.get_command("core")
<tank.deploy.tank_command.SgtkSystemCommand object at 0x108521310>
>>> cmd = tk.get_command("core")
>>> dir(cmd)
['_SgtkSystemCommand__internal_action_obj', '_SgtkSystemCommand__log', '__class__', '__delattr__', '__dict__', '__doc__', '__format__', '__getattribute__', '__hash__', '__init__', '__module__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', 'category', 'description', 'execute', 'name', 'parameters', 'set_logger']
>>> cmd.execute({})
INFO 
INFO Welcome to the Shotgun Pipeline Toolkit update checker!
INFO This script will check if the Toolkit Core API installed
INFO in /mnt/wintermute/configs/sha_is_in_the_heart
INFO is up to date.

...

INFO 
{'status': 'updated', 'new_version': 'v0.17.8'}

```